api/v1alpha1/proxydefaults_types.go

```diff
@@ -123,15 +124,16 @@ func (in *ProxyDefaults) ToConsul() api.ConfigEntry {
 }
 
 func (in *ProxyDefaults) MatchesConsul(candidate api.ConfigEntry) bool {
-	proxyDefCand, ok := candidate.(*capi.ProxyConfigEntry)
+	configEntry, ok := candidate.(*capi.ProxyConfigEntry)
 	if !ok {
 		return false
 	}
-	proxyDefCand.Namespace = ""
-	proxyDefCand.CreateIndex = 0
-	proxyDefCand.ModifyIndex = 0
+	// Zero out fields from consul that we don't want to compare on.
+	configEntry.Namespace = ""
+	configEntry.CreateIndex = 0
+	configEntry.ModifyIndex = 0
```

Comment:
Do you think we could use `cmpopts.IgnoreFields` instead? Otherwise, we're modifying the `candidate` param since it's a pointer, and I believe we're later trying to read `ModifyIndex` in the controller.

Reply:
TIL that was an option. That was an excellent suggestion!!! Done!